crates/api_common/src/build_response.rs

```diff
@@ -80,14 +80,9 @@ pub async fn build_post_response(
   let is_mod_or_admin = is_mod_or_admin(&mut context.pool(), person, community_id)
     .await
     .is_ok();
-  let post_view = PostView::read(
-    &mut context.pool(),
-    post_id,
-    Some(person.id),
-    is_mod_or_admin,
-  )
-  .await?
-  .ok_or(LemmyErrorType::CouldntFindPost)?;
+  let post_view = PostView::read(&mut context.pool(), post_id, None, is_mod_or_admin)
```

Comment:
Using `None` here will cause problems, such as making the post like API return an error for local-only communities. `person: &Person` should be changed to `local_user: &LocalUser`.

Reply:
Good catch, updated. Hopefully tests will pass now.